packages/react-button/src/components/Button/Button.types.tsx

```diff
@@ -73,13 +73,16 @@ export interface ButtonProps extends ComponentProps, React.HTMLAttributes<HTMLBu
 export interface ButtonState extends ButtonProps {}
 
 export interface ButtonSlots {
+  root: React.ElementType;
   icon: React.ElementType;
   content: React.ElementType;
   loader: React.ElementType;
 }
 
 export type ButtonSlotProps = {
-  [key in keyof ButtonSlots]: ButtonProps[key];
+  [key in keyof Omit<ButtonSlots, 'root'>]: ButtonProps[key];
+} & {
+  root: React.ButtonHTMLAttributes<HTMLButtonElement>;
 };
```

Comment:
We could abstract the `root` typing into `compose` and extend it.

Reply:
I added shared typing in `react-compose`, but m not very sure that's the right package to put